core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/plugins/ai-image-prompt.plugin.ts

```diff
@@ -104,21 +123,28 @@ export class AIImagePromptView {
             .subscribe((contentlets) => {
                 const data = Object.values(contentlets[0])[0];
 
-                this.editor
-                    .chain()
-                    .deleteSelection()
-                    .insertImage(data)
-                    .openAIContentActions(DOT_AI_IMAGE_CONTENT_KEY)
-                    .run();
+                const loaderNodes = findNodeByType(this.editor, NodeTypes.LOADER);
+
+                //Trust in this property to identify the image as a placeholder, until the user accept the content.
+                data[AI_IMAGE_PLACEHOLDER_PROPERTY] = true;
+
+                if (loaderNodes) {
+                    this.editor
+                        .chain()
+                        .deleteRange({ from: loaderNodes[0].from, to: loaderNodes[0].to })
+                        .insertImage(data, loaderNodes[0].from)
+                        .openAIContentActions(DOT_AI_IMAGE_CONTENT_KEY)
```

Comment:
We can centralize all Extension Keys in a Map. What are your thoughts, @hmoreras and @oidacra?

**P.S:** Not for this PR though.